cmd/config/config.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+  "fmt"
+  "os"
+  "strings"
+
+  "github.com/Originate/gt/cmd/util"
+)
+
+
+func GetMainBranch() string {
+  return getConfigurationValue("main-branch-name")
+}
+
+
+func GetParentBranch(branchName string) string {
+  return getConfigurationValue(fmt.Sprintf("%s.parent", branchName))
+}
+
+
+func GetPerennialBranches() []string {
+  return strings.Split(getConfigurationValue("perennial-branch-names"), " ")
+}
+
+
+func GetPullBranchStrategy() string {
+  pullBranchStrategy := getConfigurationValue("pull-branch-strategy")
+  if pullBranchStrategy == "" {
+    pullBranchStrategy = "rebase"
+  }
+  return pullBranchStrategy
+}
+
+
+func GetRemoteUrl() string {
+  if os.Getenv("GIT_TOWN_ENV") == "test" {
+    mockRemoteUrl := getConfigurationValue("testing.remote-url")
+    if mockRemoteUrl != "" {
+      return mockRemoteUrl
+    }
+  }
+  return util.GetCommandOutput([]string{"git", "remote", "get-url", "origin"})
+}
+
+
+func IsFeatureBranch(branchName string) bool {
+  return branchName != GetMainBranch() && !IsPernnialBranch(branchName)
+}
+
+
+func IsPernnialBranch(branchName string) bool {
+  perennialBranches := GetPerennialBranches()
+  return util.Contains(perennialBranches, branchName)
+}
+
+
+func HasRemote() bool {
+  return GetRemoteUrl() != ""
+}
+
+
+func HasRemoteUpstream() bool {
+  return util.GetCommandOutput([]string{"git", "remote", "get-url", "upstream"}) != ""
+}
+
+
+func ShouldHackPush() bool {
+  hackPushFlag := getConfigurationValue("hack-push-flag")
+  if hackPushFlag == "" {
+    hackPushFlag = "true"
```

Comment:
This function does return a bool. The possible values we store in the config are "true" and "false" because we have to store strings. This is just giving the default.